agent/agentssh/agentssh.go

```diff
@@ -131,14 +131,15 @@ func NewServer(ctx context.Context, logger slog.Logger, prometheusRegistry *prom
 	// Clients' should ignore the host key when connecting.
 	// The agent needs to authenticate with coderd to SSH,
 	// so SSH authentication doesn't improve security.
-	randomHostKey, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		return nil, err
-	}
-	randomSigner, err := gossh.NewSignerFromKey(randomHostKey)
+
+	// Create a fixed host key, as at least one host key has to
+	// exist for the SSH server to start. We'll later replace
+	// the host key with one that's based off the workspace uuid.
+	coderSigner, err := coderSigner(42)
```

Comment:
Do we actually need a key at this time? I don't see the point of creating one if, by design, we replace it before anyone can possibly connect.